azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -71,8 +80,20 @@
   private ConcurrentHashMap<Integer, ExecutableFlow> recentlyFinished =
       new ConcurrentHashMap<Integer, ExecutableFlow>();
 
-  private ExecutingManagerUpdaterThread executingManager;
+  /* map to easily access queued flows */
+  final private ConcurrentHashMap<Integer, Pair<ExecutionReference, ExecutableFlow>> queuedFlowMap =
```

Comment:
Nice suggestion. Refactored to defined a separate data structure class and corresponding tests.